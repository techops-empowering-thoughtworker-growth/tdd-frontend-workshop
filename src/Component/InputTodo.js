import React, { useState } from 'react';
import axios from 'axios';
import PropTypes from 'prop-types';

const InputTodo = (props) => {
  const [inputText, setInputText] = useState('');

  const handleAdd = () => {
    if (inputText.trim()) {
      axios
        .post('http://localhost:8080/tasks', {
          name: inputText,
          completed: false,
        })
        .then((res) => res.data)
        .then((todo) => props.setTodoItems([...props.todoItems, todo]));
      setInputText('');
    } else {
      alert('Please write item');
    }
  };

  return (
    <div className={'input-bar'}>
      <input
        className={'add-input'}
        value={inputText}
        placeholder={'Enter your todo item'}
        onChange={(e) => setInputText(e.target.value)}
      />
      <div className={`add-button`} onClick={handleAdd}>
        +
      </div>
    </div>
  );
};

InputTodo.propTypes = {
  addTodoProps: PropTypes.func,
};

export default InputTodo;
