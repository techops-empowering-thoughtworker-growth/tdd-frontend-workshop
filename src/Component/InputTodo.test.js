import { render, mount } from 'enzyme';
import axios from 'axios';
import { act } from 'react-dom/test-utils';
import InputTodo from './InputTodo';

const props = {
  todoItems: [],
  setTodoItems: jest.fn(),
};

const addResponse = {
  data: {
    id: 3,
    name: 'task 03',
    completed: false,
  },
};

jest.mock('axios');

describe('InputTodo', () => {
  it('should take InputTodo snapshot', () => {
    const wrapper = render(<InputTodo {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should add a todo item when click the add button', async () => {
    let wrapper;
    const mockAdd = axios.post.mockResolvedValue(addResponse);

    await act(async () => {
      wrapper = mount(<InputTodo {...props} />);
    });
    wrapper.update();
    wrapper
      .find('.add-input')
      .simulate('change', { target: { value: 'task 03' } });

    await act(async () => {
      wrapper.find('.add-button').simulate('click');
    });
    wrapper.update();

    expect(mockAdd).toHaveBeenCalledWith('http://localhost:8080/tasks', {
      completed: false,
      name: 'task 03',
    });
  });
});
