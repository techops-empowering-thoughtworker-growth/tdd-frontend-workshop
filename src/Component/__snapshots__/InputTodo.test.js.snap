// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputTodo should take InputTodo snapshot 1`] = `
LoadedCheerio {
  "0": <div
    class="input-bar"
  >
    <input
      class="add-input"
      placeholder="Enter your todo item"
      value=""
    />
    <div
      class="add-button"
    >
      +
    </div>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
